pkg/config/config_template.yaml

```diff
@@ -1435,6 +1435,9 @@ api_key:
   ##  * require - list of key or key/value regex pattern strings - traces must have those tags in order to be sent to Datadog
   ##  * reject - list of key or key/value regex pattern strings - traces with these tags are dropped by the Agent
   ## Note: Rules take into account the intersection of tags defined.
+  ## Using regex patterns in the tag filtering reduces performance by about 3 times.
```

Comment:
nit: I'd just say "Using regexp patterns for tag filtering can have performance implications, and be a bit slower than typical tag filtering without regexp. However, this regexp is only run on the root span of a trace, so should not have a critical impact on overall performance."

Reply:
I just went ahead and committed this directly so we can merge this PR :)